x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/rules_table_context.tsx

```diff
@@ -303,6 +355,11 @@ export const RulesTableContextProvider = ({ children }: RulesTableContextProvide
         loadingRulesAction: loadingRules.action,
         selectedRuleIds,
         sortingOptions,
+        isDefault: isDefaultState(filterOptions, sortingOptions, {
+          page,
+          perPage,
+          total: isInMemorySorting ? rules.length : total,
```

Comment:
FYI the in-memory mode has been deleted by @xcrzx in https://github.com/elastic/kibana/pull/149840